nemo/collections/tts/models/fastpitch.py

```diff
@@ -511,45 +509,29 @@ def list_available_models(cls) -> 'List[PretrainedModelInfo]':
 
         return list_of_models
 
-    # Methods for model exportability
-    def _prepare_for_export(self, **kwargs):
-        super()._prepare_for_export(**kwargs)
-
-        # Define input_types and output_types as required by export()
-        self._input_types = {
-            "text": NeuralType(('B', 'T_text'), TokenIndex()),
-            "pitch": NeuralType(('B', 'T_text'), RegressionValuesType()),
-            "pace": NeuralType(('B', 'T_text'), optional=True),
-            "speaker": NeuralType(('B'), Index()),
-        }
-        self._output_types = {
-            "spect": NeuralType(('B', 'D', 'T_spec'), MelSpectrogramType()),
-            "num_frames": NeuralType(('B'), TokenDurationType()),
-            "durs_predicted": NeuralType(('B', 'T_text'), TokenDurationType()),
-            "log_durs_predicted": NeuralType(('B', 'T_text'), TokenLogDurationType()),
-            "pitch_predicted": NeuralType(('B', 'T_text'), RegressionValuesType()),
-        }
-
-    def _export_teardown(self):
-        self._input_types = self._output_types = None
-
     @property
     def disabled_deployment_input_names(self):
         """Implement this method to return a set of input names disabled for export"""
-        disabled_inputs = set()
+        disabled_inputs = set(["durs", "spec", "attn_prior", "mel_lens", "input_lens"])
         if self.fastpitch.speaker_emb is None:
             disabled_inputs.add("speaker")
         return disabled_inputs
 
+    @property
+    def disabled_deployment_output_names(self):
+        """Implement this method to return a set of output names disabled for export"""
+        disabled_outputs = set(["attn_soft", "attn_logprob", "attn_hard", "attn_hard_dur", "pitch"])
+        return disabled_outputs
+
     @property
     def input_types(self):
-        return self._input_types
+        return self.fastpitch.input_types
 
     @property
     def output_types(self):
-        return self._output_types
+        return self.fastpitch.output_types
 
-    def input_example(self, max_batch=1, max_dim=256):
+    def input_example(self, max_batch=1, max_dim=44):
```

Comment:
@borisfom Does it matter what we set max_dim to?

Reply:
I think you want to set this to a power of 2 I would try 64  instead I think 256 probably has a high memory requirement

Reply:
I set the value of 44 as this was the value that was hardcoded prior to `nemo:1.6.2`. Empirically testing (inspired by your comment), I see that the maximal value with which export succeeds on a 16GB T4 GPU is 160, hence if a power of 2 is required then the value should be 128?

Can you, please, explain what does the parameter control in the exported model? I.e. what is the difference if one does export with `max_dim=44` vs `max_dim=64` or `max_dim=128` for that matter?

It would really help if @borisfom joined in on the conversation.

Reply:
What I am super confused about is how the 256 dimension doesn't fit on a 16gb gpu however. I would test it with 256 again and double check. Perhaps it's because I am using amp.

Reply:
Thanks for the explanation, it does go in line with my assumptions. You mention a "Fastpitch's limit is 2048 according to nvidia fast pitch repo". Which repo? [DLE](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/SpeechSynthesis/FastPitch), NeMo, [NGC](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_en_fastpitch)?

That you are able to convert with `max_input=256` leaves me perplexed, as I was not able to do so even with a 40GB A100 GPU. What is your setup?  I rechecked and using the following configuration export fails: 
- GPU T4 16GB (at the moment I can not run it on a 40GB GPU)
- nemo container built from pytorch:22.04 and this PR
- RIVA 2.0.0
- tts_en_fastpitch v1.4.0 and v1.8.1, dowloaded from (https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_en_fastpitch/version)
- commands run:  	
   1. `nemo2riva --key test tts_en_fastpitch_align.nemo --out tts_en_fastpitch.riva --max_dim=256`
   2. `python3 scripts/export.py ython3 tts_en_fastpitch_align.nemo tts_en_fastpitch.onnx --max-dim 256`

You mention that you are using amp, where? I don’t see any parameters in `nemo2riva` or `export.py` for that. If you, however, mean during training, the same error occurs on my custom models that were trained with `precision:16` and `bf16`.

Reply:
Right I don't use the nemo variant of fastpitch I use the R and D variant so it's likely configured differently. 
https://github.com/NVIDIA/DeepLearningExamples I use FP from this repo

Reply:
@itzsimpl yeah this worked for me strange I do have a 3080 but I tried it on cpu and works at 128